nmcp/src/main/kotlin/nmcp/internal/utils.kt

```diff
@@ -134,16 +134,17 @@ internal fun Project.registerPublishToCentralPortalTasks(
      * This gives feedback to the user before compiling all projects.
      */
     project.gradle.taskGraph.whenReady {
-        val extensionName = when(name) {
-            "aggregation" -> nmcpAggregationExtensionName
-            else -> nmcpExtensionName
-        }
         if (it.hasTask(taskPath(project, task.name))) {
+            val publishingType = spec.publishingType.orNull
+            val validValues = listOf("AUTOMATIC", "USER_MANAGED")
+            check(publishingType == null || publishingType in validValues) {
+                "Nmcp: invalid Central Portal publishingType value '$publishingType'. Must be one of ${validValues}."
```

Comment:
If only there was a way like data objects to prevent invalid values at the configuration time.

Reply:
AFAIK Kotlin's data objects (== `static final ... fields`) should be perfectly usable with both Java and Groovy.

For instance:

```
sealed class PublishingType {
    data object Automatic : PublishingType()
    data object UserManaged : PublishingType()
    data class Other(val value: String) : PublishingType()
}
```

Both Groovy and Java should support cases like `publishingType = PublishingType.Automatic` or `publishingType = new PublishingType.Other("some_new_value_added_by_sonatype")`

Reply:
Ah yes of course. No exhaustive `when` there but it's not an issue at all in that case and allows the `Other` case 👍

Reply:
It might be worth adding `static final` accessors to reduce `.INSTANCE` boilerplate, however, I'm not sure how common is it:

```kotlin
sealed class PublishingType {
    data object Automatic : PublishingType()
    data object UserManaged : PublishingType()
    data class Other(val value: String) : PublishingType()

    companion object {
        @JvmField val AUTOMATIC: PublishingType = Automatic
        @JvmField val USER_MANAGED: PublishingType = UserManaged
    }
}
```